File: C#/Matrix/Matrix.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Matrix
{
    List<List<int>> _data = new List<List<int>>();
    (int, int) _size;

    public Matrix(string input)
    {
        var rows = input.Split('\n');

        foreach (var row in rows)
        {
            var r = row.Split(" ").Select(int.Parse).ToList();
            _data.Add(r);
        }

        _size = (_data[0].Count, _data.Count);
    }

    public int[] Row(int row)
    {
        return _data[row-1].ToArray();
    }

    public int[] Column(int col)
    {
        return _data.Select(row => row[col - 1]).ToArray();
    }
}

File: TypeScript/matrix.ts
export class Matrix {
    readonly data: string[];

    constructor(matrix: string) {
      this.data = matrix.split('\n');
    }
  
    get rows(): number[][] {
      return this.data.map((row) => row.split(' ').map(Number));
    }
  
    get columns(): number[][] {
      const rows = this.rows;

      let xLen = rows[0].length;
      let yLen = rows.length;
      
      let columns = [];
      
      for(let x = 0; x < xLen; x++) {
          let column = [];
      
          for(let y = 0; y < yLen; y++) {
            column.push(rows[y][x]);
          }
          
          columns.push(column)
      }
      
      return columns;  
    }
}

const m: Matrix = new Matrix('1 2 3\n4 5 6');

console.log(m.rows);
console.log(m.columns);